tools/bpf2c/bpf2c_driver.c

```diff
@@ -0,0 +1,122 @@
+// Copyright (c) Microsoft Corporation
+// SPDX-License-Identifier: MIT
+
+#include <wdm.h>
+#include <wsk.h>
+
+#define BPF2C_DRIVER_CODE
+#include "bpf2c.h"
+
+DRIVER_INITIALIZE DriverEntry;
+DRIVER_UNLOAD DriverUnload;
+RTL_QUERY_REGISTRY_ROUTINE static _bpf2c_query_registry_routine;
+
+static NPI_MODULEID _bpf2c_module_id = {sizeof(_bpf2c_module_id), MIT_GUID, {0}};
+static HANDLE _bpf2c_nmr_client_handle;
+static HANDLE _bpf2c_nmr_provider_handle;
+metadata_table_t ___META_DATA_TABLE___;
+
+static NTSTATUS
+_bpf2c_npi_client_attach_provider(
+    HANDLE nmr_binding_handle, void* client_context, const NPI_REGISTRATION_INSTANCE* provider_registration_instance);
+
+static NTSTATUS
+_bpf2c_npi_client_detach_provider(void* ClientBindingContext);
+
+static const NPI_CLIENT_CHARACTERISTICS _bpf2c_npi_client_characteristics = {
+    0,                                  // Version
+    sizeof(NPI_CLIENT_CHARACTERISTICS), // Length
+    _bpf2c_npi_client_attach_provider,
+    _bpf2c_npi_client_detach_provider,
+    NULL,
+    {0,                                 // Version
+     sizeof(NPI_REGISTRATION_INSTANCE), // Length
+     &bpf2c_npi_id,
+     &_bpf2c_module_id,
+     0,
+     &___META_DATA_TABLE___}};
+
+NTSTATUS
+bpf2c_query_npi_module_id(
+    PWSTR ValueName, ULONG ValueType, PVOID ValueData, ULONG ValueLength, PVOID Context, PVOID EntryContext)
```

Comment:
Can we use SAL annotation in this file?  Also follow [coding style guidance (don't start arg names with capital letters).](https://github.com/microsoft/ebpf-for-windows/blob/master/docs/DevelopmentGuide.md#naming-conventions), e.g., `lower_snake_case` not `PascalCase`, etc.